lib/InfluxRequest.js

```diff
@@ -106,20 +107,23 @@ InfluxRequest.prototype._request = function (options, callback) {
   // need to store the original path, in case we need to re-submit the request onError and change the host
   if (!requestOptions.originalUrl) requestOptions.originalUrl = requestOptions.url
   requestOptions.url = this.url(host, requestOptions.originalUrl)
-  requestOptions.retries++
-  request(requestOptions, function (err, response, body) {
-    self._parseCallback(err, response, body, requestOptions, callback)
-  })
-}
-
-InfluxRequest.prototype._parseCallback = function (err, response, body, requestOptions, callback) {
-  if (err && resubmitErrorCodes.indexOf(err.code) !== -1) {
-    this.disableHost(requestOptions.host)
-    if (this.options.maxRetries >= requestOptions.retries && this.hostIsAvailable()) {
-      return this._request(requestOptions, callback)
+  requestOptions.retries++  // fetch() ignores the `.retries` option
+  var theResponse
+  fetch(requestOptions.url, requestOptions).then(function (response) {
+    theResponse = response
+    return response.text()
+  }).then(function (body) {
+    body = body === '' ? {} : JSON.parse(body)  // workaround for https://github.com/bitinn/node-fetch/issues/165
```

Comment:
That would be less resilient to imperfect implementations that return, say, 200 and an empty response, but it would help catch that error.

I've amended my [PR to node-fetch to check for a status code of 204 instead](https://github.com/bitinn/node-fetch/pull/166/commits/a6527028b15cac83002a7275b9a2bc3d8024f657).